pkg/asset/machines/worker.go

```diff
@@ -178,7 +178,7 @@ func (w *Worker) Generate(dependencies asset.Parents) error {
 			}
 		case azuretypes.Name:
 			mpool := defaultAzureMachinePoolPlatform()
-			mpool.InstanceType = azuredefaults.InstanceClass(installconfig.Config.Platform.Azure.Region)
+			mpool.InstanceType = azuredefaults.InstanceClassByNodeType(installconfig.Config.Platform.Azure.Region, "worker")
```

Comment:
`pkg/machines/worker.go` is the authority for what instance to choose for workers. `InstanceClassByNodeType` moves that control to `pkg/types/azure`

So personally i lean towards what we do for AWS [`awsDefaultWorkerMachineType`](https://github.com/openshift/installer/pull/1868/files#diff-df3202277ffea159153362ec095a8762R107) being called [here](https://github.com/openshift/installer/pull/1868/files#diff-df3202277ffea159153362ec095a8762R136)

Reply:
I have reworked this to be more in line with your thinking here.